--- src/torrent/download_old.cc	2014-05-14 22:18:48.000000000 +0800
+++ src/torrent/download.cc	2017-08-22 23:09:22.000000000 +0800
@@ -582,8 +582,8 @@
 
 void
 Download::add_peer(const sockaddr* sa, int port) {
-  if (m_ptr->info()->is_private())
-    return;
+  //if (m_ptr->info()->is_private())
+  //  return;
 
   rak::socket_address sa_port = *rak::socket_address::cast_from(sa);
   sa_port.set_port(port);
