--- src/command_download_old.cc	2017-08-22 23:12:48.000000000 +0800
+++ src/command_download.cc	2017-08-22 23:17:21.000000000 +0800
@@ -305,8 +305,8 @@
   char dummy;
   char host[1024];
 
-  if (download->download()->info()->is_private())
-    throw torrent::input_error("Download is private.");
+  //if (download->download()->info()->is_private())
+  //  throw torrent::input_error("Download is private.");
 
   ret = std::sscanf(arg.c_str(), "%1023[^:]:%i%c", host, &port, &dummy);
 
@@ -319,6 +319,8 @@
     throw torrent::input_error("Invalid port number.");
 
   torrent::connection_manager()->resolver()(host, (int)rak::socket_address::pf_inet, SOCK_STREAM, call_add_d_peer_t(download, port));
+
+  lt_log_print(torrent::LOG_PEER_INFO, "Peer %s:%i added by manually.", host, port);
 }
 
 torrent::Object
